Support query by name

dpino opened the pull request:
Allows to select a snabb instance by name in `lwaftr query`.

Before `lwaftr query` had an ad-hoc implementation that allowed to select a Snabb instance by name. Only SnabbVMX used this mechanism. When SnabbVMX was called passing the --id argument, it stored an instance id in a counter called 'nic/id'.

A PID in `lwaftr query` could be either a PID or an SnabbVMX's ID. In case of being an ID, its value was compared to the value stored in 'nic/id'.

This PR adds proper support for querying by name using the standard mechanism (search instance name in `/by-name` folder). Additionally, the PR reworks other subcommands that used the former implementation.

Fixes #623.

dpino committed:
Redo by-name query support in 'lwaftr query'

dpino committed:
Redo by-name query support in 'snabbvmx query'

dpino committed:
Redo by-name query support in 'snabbvmx top'

dpino committed:
Redo by-name query support in 'lwaftr monitor'

tsyesika commented on src/apps/lwaftr/lwutil.lua:
```diff
@@ -116,3 +117,13 @@ function nic_exists(pci_addr)
       dir_exists(("%s/0000:%s"):format(devices, pci_addr))
 end
+
+function get_pid_by_name (name)
```
Take a look at https://github.com/Igalia/snabb/blob/lwaftr/src/core/app.lua#L165 - It does a smiliar thing and gives you back a table of the named processes with the key being the name and the value being the PID.

tsyesika commented on src/program/snabbvmx/query/query.lua:
```diff
@@ -98,15 +98,22 @@ local function print_counters (pid, dir)
 
+local named_program_root = engine.named_program_root
+
+function get_name_by_pid (pid)
```
This can also be done with https://github.com/Igalia/snabb/blob/lwaftr/src/core/app.lua#L165 - just pass "true" as the first argument nd it'll give you back a table with pid's as the name.

It would be a bit slower than this approach though. If that's an important issue maybe this should be put in app.lua with the other name stuff it seems generic and useful enough.

tsyesika reviewed (changes requested):
Few little nits

dpino committed:
Cast pid argument to a string

List of instances are pid strings. The pid passed as an argument could
be either a string or a number. Force cast to string to guarantee values
can be successfully compared.

dpino committed:
Replace functions 'get_pid_by_name' and 'get_name_by_pid'

dpino commented:
@xray7224 Fixed issues. The last two commits contain the relevant changes.

tsyesika reviewed (approved):
LGTM, thanks for the nit fixes.
